lib/stdlib/src/maps.erl

```diff
@@ -220,14 +225,37 @@ remove(_,_) -> erlang:nif_error(undef).
 
 take(_,_) -> erlang:nif_error(undef).
 
--spec to_list(Map) -> [{Key,Value}] when
-    Map :: #{Key => Value}.
+-spec to_list(MapOrIterator) -> [{Key, Value}] when
+    MapOrIterator :: #{Key => Value} | iterator(Key, Value).
 
 to_list(Map) when is_map(Map) ->
     to_list_internal(erts_internal:map_next(0, Map, []));
+to_list({_K, _V, _NextIterator} = Iterator) ->
+    try to_list_from_iterator(Iterator) of
+        Result -> Result
+    catch
+        error:badarg ->
+            badarg_with_info([Iterator])
```

Comment:
Regarding the failed test case, the tuple was confused with an iterator. @sverker  and I discussed it and we'll think that the most logical is to raise a `badmap` exception for bad iterators too. I think that the following (not tested) will work:

```suggestion
            error_with_info(badmap, [Iterator])
```